app-backend/api/src/main/scala/uploads/Routes.scala

```diff
@@ -62,9 +73,25 @@ trait UploadRoutes extends Authentication
   }
 
   def createUpload: Route = authenticate { user =>
+    implicit def stringAsJavaURI(uri: String): URI = new URI(uri)
     entity(as[Upload.Create]) { newUpload =>
       authorize(user.isInRootOrSameOrganizationAs(newUpload)) {
-        onSuccess(write[Upload](Uploads.insertUpload(newUpload, user))) { upload =>
+        val uploadToInsert = (newUpload.uploadType.toString, newUpload.source) match {
+          case ("S3", Some(source)) => {
+            val files = S3.getObjectPaths(source).filter { p =>
```

Comment:
This process that gets the files and then filters them down could probably be separated into the `package.scala` file. Is that a good idea?